### PR TITLE
feat(examples): add MobileNetV1 CIFAR-10 training epoch validation

### LINKED ISSUE #5526: Validate one CIFAR-10 training epoch shows decreasing loss
Using the implementation from the prior task, run one training epoch of the MobileNetV1 CIFAR-10 script on real CIFAR-10 data inside the container (just shell then just test-mojo / the example's train entrypoint, since the Mojo runtime needs GLIBC 2.32+ and may not run on the host). A bounded subset of the training set is acceptable if a full epoch exceeds 60 minutes on CPU — document the subset size if used. Capture the epoch log demonstrating batch loss decreasing over the epoch. If loss fails to decrease or produces NaN/inf, diagnose via the smoke test and hand corrections back before proceeding. Save the captured log for the PR.

_Registered from Telemachy workflow `mobilenetv1-backward-pass`; assign_to `task-agent`._
Depends on #5525

### PR DESCRIPTION
**Pull request by mvillmow**
## Summary
Add scripts to validate one training epoch of MobileNetV1 on CIFAR-10 with real data, demonstrating decreasing loss over batches. Includes epoch runner and log summarizer for capturing training progress.

## Changes
- Add run_mobilenetv1_cifar10_epoch.sh to execute one training epoch with configurable batch size and data subset
- Add summarize_epoch_log.py to parse and summarize training logs, extracting batch loss trends and epoch statistics

## Testing
- Run one CIFAR-10 training epoch inside container (just shell → just test-mojo or example entrypoint)
- Verify batch loss decreases monotonically over the epoch
- Verify no NaN/inf values in loss output
- Capture epoch log demonstrating loss convergence

## Closes
Closes #5526

Generated by Claude Code via ProjectHephaestus automation.

### CODE REVIEW

File: scripts/run_mobilenetv1_cifar10_epoch.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Validation runner for issue #5526. Untracked artifacts under logs/ (gitignored).
+set -uo pipefail   # NOTE: no -e — we need the Mojo exit code, not a shell abort.
+
+DATE=$(date +%F)
+LOG="logs/mobilenetv1-cifar10-epoch-${DATE}.log"
+mkdir -p logs
+
+# Helper function to run command in container (non-interactive)
+run_in_container() {
+    docker compose exec -T projectodyssey-dev bash -c "cd /workspace && $*"
```
Comment (mvillmow) [inline scripts/run_mobilenetv1_cifar10_epoch.sh]: Uses `docker compose exec` but this repo's dev environment is **podman**, not Docker. The justfile (line 8) defines `podman_service := "projectodyssey-dev"` and all justfile recipes use `podman compose`. CLAUDE.md §Troubleshooting also mandates `just podman-up`. The plan explicitly specified `podman compose exec -T ... bash -lc '...'`. On a podman-only host, `docker` either does not exist or points to a different daemon, so this invocation fails at the `run_in_container` call before any training runs. Fix: replace `docker compose exec -T projectodyssey-dev` with `podman compose exec -T projectodyssey-dev`.

Comment (mvillmow) [inline scripts/run_mobilenetv1_cifar10_epoch.sh]: In-container detection branch from the approved plan was dropped. The plan specified detecting `/.dockerenv`/`$container`/cgroup to bypass the double-exec when already inside the container. Without it, running this script from `just shell` (inside the container) will attempt `docker/podman compose exec` from within the container — which either fails (no compose available) or double-nests. Fix: add `if [ -f /.dockerenv ] || [ -n "${container:-}" ]; then bash -lc "$*"; else podman compose exec -T projectodyssey-dev bash -lc "$*"; fi` in `run_in_container`.

File: scripts/run_mobilenetv1_cifar10_epoch.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    echo "# Command: pixi run mojo run -I src -I . examples/mobilenetv1_cifar10/train.mojo --epochs 1 --batch-size 128 --lr 0.01 --data-dir datasets/cifar10"
+    echo "# Subset: full epoch (num_batches = ceil(50000/128) = 391)"
+    echo "# Started: $(date -Iseconds)"
+    echo "# ---"
+} > "$LOG"
+
+# Ensure dataset is present (CIFAR-10 is ~170MB; download once).
+if [ ! -d "datasets/cifar10" ] || [ -z "$(ls -A datasets/cifar10 2>/dev/null)" ]; then
+    echo "# Downloading CIFAR-10 to datasets/cifar10/..." | tee -a "$LOG"
+    run_in_container "pixi run python3 examples/mobilenetv1_cifar10/download_cifar10.py cifar10 datasets/cifar10" 2>&1 | tee -a "$LOG"
```
Comment (mvillmow) [inline scripts/run_mobilenetv1_cifar10_epoch.sh]: Download invocation uses positional args `cifar10 datasets/cifar10` but the plan's verified form was `--output-dir datasets/cifar10`. The `download_cifar10.py` wrapper delegates to `hephaestus.datasets.downloader.main` whose CLI signature is not positional — calling it with two positional args will likely fail or silently use wrong paths, skipping the download. Verify the actual CLI interface of `hephaestus.datasets.downloader.main` and use keyword flags.

**Review by mvillmow — COMMENTED** [review]
Critical: docker vs podman mismatch breaks all container invocations. Major: missing in-container detection and wrong download CLI args. NOGO until docker→podman fix applied.

File: scripts/run_mobilenetv1_cifar10_epoch.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+# Header (gitignored file, so free to include full command line + env).
+{
+    echo "# MobileNetV1 CIFAR-10 — one-epoch validation for issue #5526"
+    echo "# Date: ${DATE}"
+    echo "# Command: pixi run mojo run -I src -I . examples/mobilenetv1_cifar10/train.mojo --epochs 1 --batch-size 128 --lr 0.01 --data-dir datasets/cifar10"
+    echo "# Subset: full epoch (num_batches = ceil(50000/128) = 391)"
+    echo "# Started: $(date -Iseconds)"
+    echo "# ---"
```
Comment (mvillmow) [inline scripts/run_mobilenetv1_cifar10_epoch.sh:27]: Re-opening: prior review comment not addressed — Line 33 still passes two bare positional args (`cifar10 datasets/cifar10`) to download_cifar10.py. No keyword flag (--output-dir or otherwise) was added and no evidence the CLI accepts positional args was provided.

> Download invocation uses positional args `cifar10 datasets/cifar10` but the plan's verified form was `--output-dir datasets/cifar10`. The `download_cifar10.py` wrapper delegates to `hephaestus.datasets.downloader.main` whose CLI signature is not positional — calling it with two positional args will likely fail or silently use wrong paths, skipping the download. Verify the actual CLI interface of `hephaestus.datasets.downloader.main` and use keyword flags.

**Review by mvillmow — COMMENTED** [review]
Re-opening 1 prior review comment(s) the current diff does not address.

File: scripts/run_mobilenetv1_cifar10_epoch.sh [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+} > "$LOG"
+
+# Ensure dataset is present (CIFAR-10 is ~170MB; download once).
+if [ ! -d "datasets/cifar10" ] || [ -z "$(ls -A datasets/cifar10 2>/dev/null)" ]; then
+    echo "# Downloading CIFAR-10 to datasets/cifar10/..." | tee -a "$LOG"
+    run_in_container "cd /workspace && pixi run python3 examples/mobilenetv1_cifar10/download_cifar10.py cifar10 datasets/cifar10" 2>&1 | tee -a "$LOG"
```
Comment (mvillmow) [inline scripts/run_mobilenetv1_cifar10_epoch.sh]: Positional args `cifar10 datasets/cifar10` are passed to `download_cifar10.py`, but the plan (verified against `hephaestus.datasets.downloader.main`) specifies the keyword form `--output-dir datasets/cifar10`. If the downloader does not accept positional args, the download fails and training exits non-zero (misreported as `TRAINING_FAILURE` rather than a script bug). Fix: `pixi run python3 examples/mobilenetv1_cifar10/download_cifar10.py --output-dir datasets/cifar10`.

**Review by mvillmow — COMMENTED** [review]
Download invocation uses positional args instead of verified --output-dir keyword; blocks successful epoch validation. All other changes (podman fix, in-container detection, summarizer) are correct.

**Review by mvillmow — COMMENTED** [review]
Correct and complete validation runner. Non-blocking: (1) `projectodyssey-dev` service name hardcoded at line 15 vs plan's `${PODMAN_SERVICE:-dev}`; (2) `float(m.group(3))` at summarize_epoch_log.py:34 has no ValueError guard for non-numeric regex matches.
